http-clients/src/test/java/org/triplea/http/client/web/socket/WebSocketConnectionTest.java

```diff
@@ -52,39 +55,71 @@ void tearDown() {
 
     @Test
     void onMessage() {
-      webSocketConnection.getClient().onMessage(MESSAGE);
+      webSocketConnection.getWebSocketListener().onText(mock(WebSocket.class), MESSAGE, true);
```

Comment:
Have you considered making the websocket listener a stand-alone class? This kind of test is a smell, (law of demeter violation). It we simpliy injected the listener implementation we might be able to decompose the test cases more thoroughly.

Reply:
I did consider it, but the current implementation is tied too much to the WebSocketConnection class.
I cleaned up the test cases a bit though